src/shared/dissect-image.c

```diff
@@ -3165,18 +3165,26 @@ int verity_dissect_and_mount(
         _cleanup_(loop_device_unrefp) LoopDevice *loop_device = NULL;
         _cleanup_(dissected_image_unrefp) DissectedImage *dissected_image = NULL;
         _cleanup_(verity_settings_done) VeritySettings verity = VERITY_SETTINGS_DEFAULT;
+        bool relax_extension_release_check = false;
         DissectImageFlags dissect_image_flags;
         int r;
 
         assert(src);
         assert(dest);
 
+        if (options)
+                relax_extension_release_check =
+                        string_contains_word(mount_options_from_designator(options, PARTITION_ROOT), ",", "x-systemd.extension-release.nonstrict") ||
+                        string_contains_word(mount_options_from_designator(options, PARTITION_USR), ",", "x-systemd.extension-release.nonstrict") ||
+                        string_contains_word(options->options, ",", "x-systemd.extension-release.nonstrict");
```

Comment:
`x-systemd.relax-extension-release-check`?

Reply:
I'd prefer if this would take a boolean btw. (at least optionally), i.e `x-systemd.relax-extension-release-check=1` or so.

Also, this expression is super complex. Please add helper for it `mount_options_relax_extension_release_check()` or so that hides it

Reply:
you are right. ignore me then.

Reply:
(still: please split this out into a helper func)